pkg/segment/writer/metrics/metricssegment.go

```diff
@@ -145,11 +147,17 @@ Every 5s, this metrics buffer should persist to disk and will create / update tw
  2. TSID offset file. Format [tsid][soff]
 */
 type MetricsBlock struct {
-	tsidLookup     map[uint64]int
-	allSeries      []*TimeSeries
-	sortedTsids    []uint64
-	mBlockSummary  *structs.MBlockSummary
-	blkEncodedSize uint64 // total encoded size of the block
+	tsidLookup      map[uint64]int
+	allSeries       []*TimeSeries
+	sortedTsids     []uint64
+	mBlockSummary   *structs.MBlockSummary
+	blkEncodedSize  uint64 // total encoded size of the block
+	currentWal      *wal.WAL
+	walDPBuffer     []wal.WalDatapoint
+	allWALFD        []*wal.WAL
+	currentWALIndex uint64
+	segID           uint64
+	shardID         string
```

Comment:
for now keep the name of this as mId, we are going to do a renaming of mid once everywhere later